Wikipedia/Code/UIImageView+WMFImageFetchingInternal.m

```diff
@@ -187,6 +190,14 @@ - (void)wmf_setImage:(UIImage *)image
     };
     
     self.image = image;
+    
+    if (image.isGIF && [self isKindOfClass:[FLAnimatedImageView class]]) {
+        NSURL *imageURL = [self wmf_imageURLToFetch];
+        FLAnimatedImage *animatedImage = [FLAnimatedImage animatedImageWithGIFData: [[self wmf_imageController] diskDataForImageWithURL:imageURL]];
```

Comment:
Actually, instead you could add the image data to the `WMFImageDownload` returned by `fetchImageWithURL:...` in `wmf_fetchImageDetectFaces:...` - then pass that data to this method to use rather than re-querying the image manager for `diskDataForImageWithURL`. It should still be nil-checked. Passing the data in this way also avoids a potentially costly disk read on the main thread.